src/choreo/voter/fd_voter.c

```diff
@@ -0,0 +1,154 @@
+#include "fd_voter.h"
+#include <string.h>
+
+#pragma GCC diagnostic ignored "-Wformat"
+#pragma GCC diagnostic ignored "-Wformat-extra-args"
+
+ulong
+fd_vote_txn_generate(fd_voter_t                       voter,
+                     fd_compact_vote_state_update_t * vote_update,
+                     uchar *                          recent_blockhash,
+                     uchar out_txn_meta_buf [static FD_TXN_MAX_SZ],
+                     uchar out_txn_buf [static FD_TXN_MTU]
+                     ) {
+  fd_pubkey_t node_pubkey, authorized_voter_pubkey, vote_program_pubkey;
+  memcpy( node_pubkey.key, voter.node_keypair + 32UL, 32 );
+  memcpy( authorized_voter_pubkey.key, voter.authorized_voter_keypair + 32UL, 32 );
+  memcpy( vote_program_pubkey.key, &fd_solana_vote_program_id, sizeof(fd_pubkey_t) );
```

Comment:
why memcpy here ?

Reply:
Without `memcpy`, I will need to do the `(fd_pubkey_t *) fd_type_pun_const( ... )` business in my latest push. Please take a look a see whether you find those type casts fine.